tests/e2e/index_auto_creation/index_auto_creation.py

```diff
@@ -0,0 +1,318 @@
+# Copyright 2024 Memgraph Ltd.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+# License, and you may not use this file except in compliance with the Business Source License.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0, included in the file
+# licenses/APL.txt.
+
+import sys
+
+import pytest
+from common import cursor, execute_and_fetch_all
+
+
+# Helper functions
+def get_index_stats(cursor):
+    return execute_and_fetch_all(cursor, "SHOW INDEX INFO")
+
+
+def index_exists(indices, index_name):
+    for index in indices:
+        if index[1] == index_name:
+            return True
+
+    return False
+
+
+def index_count_is(indices, index_name, count):
+    return count == index_count(indices, index_name)
+
+
+def number_of_index_structures_are(indices, predicted_size):
+    return len(indices) == predicted_size
+
+
+def index_count(indices, index_name):
+    for index in indices:
+        if index[1] == index_name:
+            return index[3]
+
+    return 0
+
+
+#####################
+# Label index tests #
+#####################
+
+
+def test_auto_create_single_label_index(cursor):
+    label = "SOMELABEL"
+    execute_and_fetch_all(cursor, f"CREATE (n:{label})")
+    index_stats = get_index_stats(cursor)
+    assert number_of_index_structures_are(index_stats, 1)
+    assert index_exists(index_stats, label)
+    assert index_count_is(index_stats, label, 1)
+
+    execute_and_fetch_all(cursor, f"DROP INDEX ON :{label}")
+    index_stats = get_index_stats(cursor)
+    assert number_of_index_structures_are(index_stats, 0)
+
+    execute_and_fetch_all(cursor, f"MATCH (n) DETACH DELETE n")
+
+
+def test_auto_create_multiple_label_index(cursor):
```

Comment:
Can we add test when we assign multiple labels to one vertex?

Reply:
Added.